File: src/main/java/io/gaegul/ch01/puzzle/Queen.java
package io.gaegul.ch01.puzzle;

import io.gaegul.ch01.puzzle.behavior.BowAndArrowBehavior;

public class Queen extends Character {

	protected Queen() {
		super(new BowAndArrowBehavior());
	}

	@Override
	public void fight() {
		System.out.println("여왕은 ");
		this.performWeapon();
		System.out.println("으로 전투합니다.");
	}
}

File: src/main/kotlin/io/gaegul/ch01/puzzle/Queen.kt
package io.gaegul.ch01.puzzle

import io.gaegul.ch01.puzzle.behavior.BowAndArrowBehavior

class Queen: Character(
    BowAndArrowBehavior()
) {
    override fun fight() {
        print("여왕은 ")
        weaponBehavior.useWeapon()
        print("으로 전투할 수 있다.")
    }
}